cmd/common/snapshot/process.go

```diff
@@ -35,6 +38,85 @@ type ProcessParser struct {
 	outputConfig *commonutils.OutputConfig
 }
 
+type nodeEvent struct {
+	event    *types.Event
+	children []*nodeEvent
+	parent   *nodeEvent
+}
+
+func createTree(processes []*types.Event) *nodeEvent {
+	nodes := make(map[int]*nodeEvent, 0)
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeEvent{
+			event:    process,
+			children: make([]*nodeEvent, 0),
+			parent:   nil,
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		for _, n := range nodes {
+			if node.event.Pid == n.event.ParentPid {
+				node.children = append(node.children, n)
+				n.parent = node
+			}
+		}
+	}
+
+	// Find the root, i.e. find the node without parent.
+	for _, node := range nodes {
+		if node.parent == nil {
+			return node
+		}
+	}
+
+	return nil
+}
+
+func printNode(node *nodeEvent, front string) {
+	line := fmt.Sprintf("|-%s(%d)", node.event.Command, node.event.Pid)
+	printedLine := front + line
+	if len(node.children) == 0 {
+		fmt.Println(printedLine)
+
+		return
+	}
+
+	printedLine += "-+"
+	// We add one here so the '+' and '|' of the children are aligned.
+	front += strings.Repeat(" ", len(line)+1)
+	fmt.Println(printedLine)
+
+	for _, child := range node.children {
+		printNode(child, front)
+	}
+}
+
+func printTree(tree *nodeEvent) {
+	fmt.Printf("%s (%s)\n", tree.event.Container, tree.event.Namespace)
```

Comment:
IMO, we should use a different format for the container information from the PID information. I suggest changing `container (namespace)` to `container/pod/namespace` and only printing the `/pod` and `/namespace` if available. In this way, it will be compatible with non-Kubernetes containers in local-gadget.